tokenserver/scripts/purge_old_records.py

```diff
@@ -66,19 +66,39 @@ def purge_old_records(config_file, grace_period=-1, max_per_loop=10,
                 }
                 rows = list(backend.get_old_user_records(service, **kwds))
                 logger.info("Fetched %d rows at offset %d", len(rows), offset)
+                counter = 0
                 for row in rows:
                     # Don't attempt to purge data from downed nodes.
                     # Instead wait for them to either come back up or to be
                     # completely removed from service.
                     if row.node is None:
                         logger.info("Deleting user record for uid %s on %s",
                                     row.uid, row.node)
-                        backend.delete_user_record(service, row.uid)
+                        if not settings.dryrun:
+                            backend.delete_user_record(service, row.uid)
                     elif not row.downed:
                         logger.info("Purging uid %s on %s", row.uid, row.node)
                         delete_service_data(config, service, row,
-                                            timeout=request_timeout)
-                        backend.delete_user_record(service, row.uid)
+                                            timeout=request_timeout,
+                                            settings=settings)
+                        logging.info(
+                            "Deleting user records {}".format(row.uid))
+                        if not settings.dryrun:
+                            backend.delete_user_record(service, row.uid)
+                        counter += 1
```

Comment:
the first `if row.node is None` block doesn't inc the counter but deletes -- probably should just inc it once outside of that whole if else block instead since all the cases issue a delete.

Reply:
Yeah, I was thinking that we'd only "count" rows that resulted in a deletion rather than any rows.

Reply:
The first block calls `delete_user_record` so it should inc the counter too.